sos/plugins/kubernetes.py

```diff
@@ -59,10 +60,16 @@ def setup(self):
         if not self.check_is_master():
             return
 
+        # Red Hat
         kube_cmd = "kubectl "
         if path.exists('/etc/origin/master/admin.kubeconfig'):
             kube_cmd += "--kubeconfig=/etc/origin/master/admin.kubeconfig"
 
+        # Ubuntu Charmed Distribution of Kubernetes
+        if path.exists('/root/cdk/kubeproxyconfig'):
```

Comment:
Isn't it better to set `self.kube_ctl` in explicit UbuntuKubernetes and RedHatKubernetes plugins?

Reply:
I think that would be less confusing - right now there's a mixture of Ubuntu and Red Hat packaging and paths in the plugin, and the path that is chosen as the key to select the Red Hat `--kubeconfig` is not listed in the plugin files list.